src/realm.h

```diff
@@ -355,35 +355,70 @@ typedef struct realm_websocket_endpoint {
     bool is_ssl;            // true if SSL should be used
 } realm_websocket_endpoint_t;
 
+// The following definitions are for internal structures and pointers used
```

Comment:
It seems the SDKs use them too. I am not a big fan splitting it this way, but I don't have a strong opinion either. People directly using this API can have their say.